flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/CheckpointBarrierUnaligner.java

```diff
@@ -387,12 +378,28 @@ synchronized void resetReceivedBarriers(long checkpointId) {
 			return allBarriersReceivedFuture;
 		}
 
-		synchronized void onChannelClosed() {
+		synchronized boolean onChannelClosed() throws IOException {
 			numOpenChannels--;
+
+			if (numBarriersReceived > 0) {
+				resetReceivedBarriers();
+				notifyAbort(new CheckpointException(CheckpointFailureReason.CHECKPOINT_DECLINED_INPUT_END_OF_STREAM));
```

Comment:
Can't we move this abort always to the caller side? Then the return value of onChannelClosed is more like `shouldBeAborted` and would be symmetric to `processCancellationBarrier`.

Reply:
Good thought and actually I also considered this way while refactoring. The only concern was that from the caller side, it would add `currentConsumedCheckpointId` in the abort message, but this message might not be consistent with the `currentReceivedCheckpointId` inside `ThreadSafeUnaligner`.

But I think we might remove this info from `#notifyAbort` to unify them in the caller.

Reply:
After double checking, we can not remove `checkpointId` from `#notifyAbort` method, and since the `currentReceivedCheckpointId` and `currentConsumedCheckpointId` might not be consistent always, so I think it is better to place the `#notifyAbort` in different components to get the proper id, otherwise we might confuse users to see the log info.

Reply:
Okay, I haven't considered the different checkpoint ids, so leave as is.